brave/brave-core
# Added search promotion for DDG

fix https://github.com/brave/brave-browser/issues/38692

New `BraveSearchOmniboxDDGBanner` feature is introduced to show DDG promotion banner.
As new banner types are added, p3a metric is also updated.

![Screenshot 2024-05-29 214548](https://github.com/brave/brave-core/assets/6786187/284e84b4-f5c2-43b7-870a-c2b082f4b793)
![Screenshot 2024-05-29 214610](https://github.com/brave/brave-core/assets/6786187/94a64248-9343-4591-96bb-6cda9f848a71)




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

`BraveSearchConversion*`

This promotion is disabled by default and will be controlled via griffin.
To test easily w/o griffin, we could use `--enable-features=XXX` cmd args

1. Launch brave with `--enable-features=BraveSearchOmniboxDDGBanner:banner_type/type_DDG_B`
2. Set DDG as a default search provider
3. Type anything in omnibox and check DDG promotion is shown at last
4. Set Google as a default search provider
5. Type anything in omnibox and check there is no promotion
6. Set Brave as a default search provider
7. Type anything in omnibox and check there is no promotion

As this promotion is controlled with new griffin study, user could get existing search promotion study togeter
When DDG promotion and existing promotion are enabled both,
user will see DDG promotion if current search provider is DDG.
And user will see existing promotion if current search provider is non-brave search.
To test easily this scenario, we could use `--enable-features=XXX` cmd args also.

1. Launch brave with `--enable-features=BraveSearchOmniboxBanner:banner_type/type_B,BraveSearchOmniboxDDGBanner:banner_type/type_DDG_B`
2. Set DDG as a default search provider
3. Type anything in omnibox and check DDG promotion is shown
4. Set Google as a default search provider
5. Type anything in omnibox and check previous promotion is shown
6. Set brave and check there is no promotion
